### PR TITLE
[Compression v2] Add optimizer & lr scheduler construct helper

### PR DESCRIPTION
**Pull request by J-shang**
### Description ###
Just show how to trace optimizer parameters right now. I will update this pr after #4308 #4280 

### Checklist ###
  - [x] test case
  - [x] doc

### CODE REVIEW

File: docs/en_US/Compression/v2_pruning_algo.rst [hunk 1]
```diff
@@ -155,8 +155,13 @@ Usage
 .. code-block:: python
 
    from nni.algorithms.compression.v2.pytorch.pruning import SlimPruner
+   from nni.algorithms.compression.v2.pytorch.utils import trace
```
Comment (ultmaster) [inline docs/en_US/Compression/v2_pruning_algo.rst:158]: Why is this in `algorithms`?
Suggest use another name rather than `trace` to avoid confusion.

File: nni/algorithms/compression/v2/pytorch/utils/constructor_helper.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+        return self.callable_obj(*args, **kwargs)
+
+    @staticmethod
+    def from_trace(lr_scheduler_trace: SerializableObject):
+        assert isinstance(lr_scheduler_trace, SerializableObject), \
+            'Please use nni.algorithms.compression.v2.pytorch.utils.trace to wrap the lr scheduler class before initialize the scheduler.'
+        assert isinstance(lr_scheduler_trace, _LRScheduler), \
+            'It is not an instance of torch.nn.lr_scheduler._LRScheduler.'
+        return LRSchedulerConstructHelper(lr_scheduler_trace._get_nni_attr('symbol'),
```
Comment (ultmaster) [inline nni/algorithms/compression/v2/pytorch/utils/constructor_helper.py:124]: You can use `.trace_symbol`, `.trace_args` and `.trace_kwargs`.

File: nni/algorithms/compression/v2/pytorch/pruning/movement_pruner.py [hunk 3]
```diff
@@ -171,9 +155,9 @@ def trainer(model: Module, optimizer: Optimizer, criterion: Callable[[Tensor, Te
                     # If you don't want to update the model, you can skip `optimizer.step()`, and set train mode False.
                     optimizer.step()
                 model.train(mode=training)
-    optimizer : torch.optim.Optimizer
-        The optimizer instance used in trainer. Note that this optimizer might be patched during collect data,
-        so do not use this optimizer in other places.
+    traced_optimizer : nni.common.serializer.SerializableObject(torch.optim.Optimizer)
```
Comment (ultmaster) [inline nni/algorithms/compression/v2/pytorch/pruning/movement_pruner.py:158]: There is an interface called `Traceable`. `SerializableObject` is only its concrete implementation,

File: docs/en_US/Compression/v2_pruning_algo.rst (hunk 1 above)
Comment (J-shang) [reply docs/en_US/Compression/v2_pruning_algo.rst]: `trace` -> `trace_parameters`

File: nni/algorithms/compression/v2/pytorch/utils/constructor_helper.py (hunk 2 above)
Comment (J-shang) [reply nni/algorithms/compression/v2/pytorch/utils/constructor_helper.py]: fixed

File: nni/algorithms/compression/v2/pytorch/pruning/movement_pruner.py (hunk 3 above)
Comment (J-shang) [reply nni/algorithms/compression/v2/pytorch/pruning/movement_pruner.py]: fixed

**Review by QuanluZhang — APPROVED** [review]

File: nni/algorithms/compression/v2/pytorch/utils/constructor_helper.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        assert isinstance(optimizer_trace, Optimizer), \
+            'It is not an instance of torch.nn.Optimizer.'
+        return OptimizerConstructHelper(model,
+                                        optimizer_trace._get_nni_attr('symbol'),
+                                        *optimizer_trace._get_nni_attr('args'),
+                                        **optimizer_trace._get_nni_attr('kwargs'))
+
+
+class LRSchedulerConstructHelper(ConstructHelper):
+    def __init__(self, lr_scheduler_class: Callable, *args, **kwargs):
```
Comment (ultmaster) [inline nni/algorithms/compression/v2/pytorch/utils/constructor_helper.py:99]: `Callable` -> `Type[LRScheduler]`

File: nni/algorithms/compression/v2/pytorch/utils/constructor_helper.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        self.kwargs = deepcopy(kwargs)
+
+    def call(self):
+        args = deepcopy(self.args)
+        kwargs = deepcopy(self.kwargs)
+        return self.callable_obj(*args, **kwargs)
+
+
+class OptimizerConstructHelper(ConstructHelper):
+    def __init__(self, model: Module, optimizer_class: Optimizer, *args, **kwargs):
```
Comment (ultmaster) [inline nni/algorithms/compression/v2/pytorch/utils/constructor_helper.py:37]: `Type[Optimizer]`

**Review by ultmaster — APPROVED** [review]
